pkg/operator/quorumguardcontroller/quorumguardcontroller.go

```diff
@@ -235,31 +239,3 @@ func (c *QuorumGuardController) ensureEtcdGuardPDB(ctx context.Context, recorder
 
 	return nil
 }
-
-// getMastersReplicaCount get number of expected masters statically defined by the controlPlane replicas in the install-config.
-func (c *QuorumGuardController) getMastersReplicaCount() (int32, error) {
```

Comment:
This doesn't seem like a value we should use when deciding whether and how to add members.  It seems simpler to add every master node available.

Reply:
Knowing the "expected" control plane size was a key requirement so we don't promote to more than `N+1` voting members at any given time (where `N` is the desired control-plane size).

Background from prior proposals:
- https://github.com/openshift/enhancements/pull/943/files/8ed85c3b3108633dced47b3a227c835ead3e921b#diff-8f08f89622ea85c6276166feaa7be109adaae9cf5d280b18ef20f4f218fce87cR137-R139
- https://github.com/openshift/enhancements/pull/920/files/22ad8ce2bcba65a0de70e1ed08d401227b030156#r738694187

When adding members we need to check against the max limit of learner members allowed `ETCD_EXPERIMENTAL_MAX_LEARNERS`:
https://github.com/openshift/cluster-etcd-operator/blob/f3d8db9c071c979b064063e97206f24398e9a854/pkg/etcdenvvar/etcd_env.go#L272-L299

Currently this is set to `N`. So if we simply add every new master node that appears into the etcd cluster as a learner member, then for the `N+1th` node the member add would be rejected.

I do have misgivings on using the InstallConfig's replica count as the desired masters replica count (since it doesn't seem like it was meant to be a declarative config) but the etcd-operator has been consuming it that way for a while now.